testsuite/smoke-test/src/nft_transaction.rs

```diff
@@ -19,92 +14,99 @@ impl Test for NFTTransaction {
-                    transfer_amount,
-                    sender_address,
-                    0,
-                ),
+        let mut creator = ctx.random_account();
+        ctx.create_user_account(creator.public_key()).await?;
+        ctx.mint(creator.address(), 10_000_000).await?;
+
+        let mut owner = ctx.random_account();
+        ctx.create_user_account(owner.public_key()).await?;
+        ctx.mint(owner.address(), 10_000_000).await?;
+
+        let collection_builder = ctx.transaction_factory().payload(
+            aptos_stdlib::encode_create_unlimited_collection_script_script_function(
+                hex::encode("description").into_bytes(),
+                hex::encode("collection name").into_bytes(),
+                hex::encode("uri").into_bytes(),
+            ),
+        );
+
+        let collection_txn = creator.sign_with_transaction_builder(collection_builder);
+        client.submit_and_wait(&collection_txn).await?;
+
+        let token_builder = ctx.transaction_factory().payload(
+            aptos_stdlib::encode_create_token_script_script_function(
+                hex::encode("collection name").into_bytes(),
+                hex::encode("description").into_bytes(),
+                hex::encode("token name").into_bytes(),
+                1,
+                hex::encode("uri").into_bytes(),
+            ),
+        );
+
+        let token_txn = creator.sign_with_transaction_builder(token_builder);
+        client.submit_and_wait(&token_txn).await?;
+
+        let token_num = client
+            .get_account_resources_by_type(
+                creator.address(),
+                AccountAddress::from_hex_literal("0x1").unwrap(),
+                &"Token".parse().unwrap(),
+                &"Collections".parse().unwrap(),
             )
-            .sender(sender_address)
-            .sequence_number(test_sequence_number)
-            .build();
-        assert_eq!(unsigned_txn.sender(), sender_address);
-
-        // sign the transaction with the private key
-        let signature = sender_private_key.sign(&unsigned_txn);
-        // submit the transaction
-        let txn = SignedTransaction::new(unsigned_txn, sender_public_key, signature);
-        client.submit_and_wait(&txn).await?;
+            .await?
+            .inner()[0]
+            .data
+            .as_object()
+            .unwrap()["collections"]
+            .as_object()
+            .unwrap()["data"]
+            .as_array()
+            .unwrap()[0]
+            .as_object()
+            .unwrap()["value"]
+            .as_object()
+            .unwrap()["tokens"]
+            .as_object()
+            .unwrap()["data"]
+            .as_array()
+            .unwrap()[0]
+            .as_object()
+            .unwrap()["value"]
+            .as_object()
+            .unwrap()["id"]
+            .as_object()
+            .unwrap()["creation_num"]
+            .as_str()
+            .unwrap()
+            .parse::<u64>()
+            .unwrap();
```

Comment:
really??

Reply:
can't you just data["collections"]["data"][0]["value"]["tokens"]["data"][0]...?

Reply:
that would be nice. do you know how? I don't think its feasible as rust doesn't really support anything remotely like that :)

Reply:
it wouldn't be hard to add a wrapper that returns self, and takes in an enum of index type, i.e object or arr. It's not perfect, but it's something. Alternatively can pass in the full path as a string. Or we do static type generation from the move struct :-)